### ISSUE TITLE
Ability to set static voicebridge PIN per conference / conference room in greenlight

### ISSUE THREAD

**Issue by wolmeh** [bigbluebutton/greenlight#1412]
see [#8371](https://github.com/bigbluebutton/bigbluebutton/issues/8371) in bigbluebutton issue section.

It should be possible to start a conference with static voicebridge PIN by bbb API: https://bbb.EXAMPLE.tld/bigbluebutton/api/create?name=EXAMPLE&meetingID=EXAMPLE&attendeePW=EXAMPLE&moderatorPW=EXAMPLE&voiceBridge=EXAMPLE&checksum=EXAMPLE

But it is not possible to set a static PIN in greenlight. 

The option to set a static voicebridge PIN which is automatically identical with conference room code would be even better.

The ability to set a static voicebridge PIN is important for conference invitations (so that you can tell the PIN before the day of the conference).

Kind regards

**Comment by davidpesce** [comment]
Hi @wolmeh - I have this working in one of the branches on my fork. It howerever will not work if you're using scalelite (or similar load balancers) with your bbb instances. I'm still trying to figure that part out!

**Comment by wolmeh** [comment]
Hi @davidpesce - any chances to merge your changes / additions with main branch? And how does your solution work? Is it necessary to set a static voicebridge PIN when you start a single conference or is it possible to set a static voicebridge PIN in conference room settings before you start a conference? We do not use a load balancer for now, but scalelite looks useful. Thanks for the clue.

**Comment by davidpesce** [comment]
It’s unlikely it’ll be merged into master. Mostly because of the scalelite issue. We are trying to come up with something that will work, but not sure if/when it’d be available. 

The phone # and pin are set in each room settings and stay that way until an owner changes them. So you could set it and forget it.

**Comment by wolmeh** [comment]
Thanks @davidpesce - I probably will try out your fork. Do you frequently update the branch [feat_dial_info](https://github.com/davidpesce/greenlight/tree/feat_dial_info)? The static PIN feature is a deal-breaker.

**Comment by denzs** [comment]
@davidpesce could you ellaborate a bit about the problems with scalelite?
I would love to have static voicebridgepins too and i tried to set it statically in greenlight (with scalelite) which seemed to work in that moment... Maybe i missed something?!

**Comment by davidpesce** [comment]
@wolmeh - I'm going to try and keep it updated as we're using it in production. that being said it'll never be our main fork. I also think it could use some tweaking to only pull phone numbers from an approved dropdown.

@denzs - the issue is that a phone number can only be assigned to one bbb server. With scalelite in front of multiple bbb servers, you'd never know which bbb instance you'll end up with (and therefore, wouldn't know the phone number). It'd be possible to do custom routing in scalelite, but I haven't dug that deep into the code.

**Comment by wolmeh** [comment]
@davidpesce Thanks a lot for your help. Do you provide a ready-to-use docker image for trying out your greenlight fork? Else we will have to compile it by ourselves.

**Comment by denzs** [comment]
@davidpesce Thanks a lot! I am aware of that problem.
I am playing around with https://github.com/denzs/fsconfregger at the moment..

I still have to test the registration against a real registrar and play around with it. I will update the examples and add an asterisk config example to have a complete solution, propably in the next days...

In parallel i have to check out and test your branch :)

**Comment by davidpesce** [comment]
I updated the branch to greenlight v2.5.5. I haven't tested just yet, but the merge conflicts were minimal.

**Comment by farhatahmad** [comment]
Going to be closing this. As @davidpesce mentioned, we can't implement this since it doesn't work with Scalelite. If it gets fixed there, maybe this can be revisited

**Comment by denzs** [comment]
I did not have the time to really implement and try this but maybe it will help other people...

I am looking from a SIP-integrators view and i am thinking about:
* use the same defaultDialAccessNumber on every bbb node
* set static voicebridgepins via greenlight (which is used as conference number inside of freeswitch)
  * so every room has the same dialaccessnumer and a static voicebridgepin which can be announced via Mail
* use fsconfregger on every bbb node to have freeswitch automatically send a registration to a central sipserver (like asterisk,freeswich or kamailio) if a conference is created

So you just need a central sipserver which is reachable via your defaultDialAccessNumber, there you could ask for the PIN and do the routing to the correct bbb node...

**Comment by lekker-xr** [comment]
[In this comment in Issue #1572](https://github.com/bigbluebutton/greenlight/issues/1572#issuecomment-629841741) someone posted a suggestion on how to achieve a static and predictable voicebridge PIN in greenlight without loadbalancer problems:

> Another approach could be creating the 5 digit numeric PIN as a hash of the 9 digit alphanumeric room name. This would be predictable and should be unique across loadbalanced instances, right?

Could this work?

For me/ us, the feature of predictable PIN is important.
